basic-services/ReadPropertyResult.cs

```diff
@@ -39,42 +40,103 @@ public struct ReadPropertyResult
+            }
             // switch on token type and set the fields appropriately
             switch (token.Type)
             {
                 case JTokenType.Integer:
                     NumericValue = token.Value<double>();
-                    StringValue = NumericValue.ToString(cultureInfo);
+                    StringValue = NumericValue.ToString(_CultureInfo);
                     BooleanValue = Convert.ToBoolean(NumericValue);
                     ArrayValue = null;
-
                     break;
                 case JTokenType.Float:
-                    //todo
+                    NumericValue = token.Value<double>();
+                    StringValue = NumericValue.ToString(_CultureInfo);
+                    BooleanValue = Convert.ToBoolean(NumericValue);
+                    break;
                 case JTokenType.String:
-                    // todo
+                    NumericValue = 0;
+                    StringValue = token.Value<string>();
+                    break;
                 case JTokenType.Array:
-                    // todo
+                    ProcessArray(token);
+                    break;
                 case JTokenType.Boolean:
-                    // todo
+                    if ((bool)(token) == true) {
+                        NumericValue = 1;
+                        BooleanValue = true;
+                        StringValue = Convert.ToString(BooleanValue);
+                    } else {
+                        NumericValue = 0;
+                        BooleanValue = false;
+                        StringValue = Convert.ToString(BooleanValue);
+                    }
+                    break;
+                case JTokenType.Object:
+                    // It is assumed the attribute read was the presentValue
+                    ProcessPresentValue(token);
+                    break;
                 default:
-                    StringValue = null;
-                    NumericValue = 0;
-                    ArrayValue = null;
-                    BooleanValue = false;
+                    NumericValue = 1;
+                    StringValue = "Unsupported Data Type";
                     break;
             }
-            
         }
 
-    }
-
-    
+        /// <summary>Parses a JArray and adds each item as a ReadPropertyResult.</summary>
+        private void ProcessArray(JToken token) {
+            JArray arr = JArray.Parse(token.ToString());
+            ArrayValue = new ReadPropertyResult[arr.Count];
+            int index = 0;
+            foreach(var item in arr.Children())
+            {
+                ArrayValue[index] = new ReadPropertyResult(Id, item, Attribute);
+                index++;
+            }
+            NumericValue = 0;
+            StringValue = "Array";
+        }
 
+        /// <summary>Searches the JObject for reliability and priority fields and uses the field called 'value' as value for result.</summary>
+        internal void ProcessPresentValue(JToken token) {
```

Comment:
You should ensure that Attribute is equal to "presentValue" to avoid this being accidentally called for any other attribute that happens to have an object value.